File: lib/python/examples/02_offboard_user.py
#!/usr/bin/env python3
"""
02_offboard_user.py — Delete all peers for a user, then delete the user account.

Usage:
  WG_BASE_URL=http://... WG_ADMIN_USER=... WG_API_TOKEN=... python 02_offboard_user.py
"""
import os
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parent.parent))
from wg_portal_client import WgPortalClient, WgPortalError

CLIENT = WgPortalClient(
    base_url=os.getenv("WG_BASE_URL", "http://192.168.1.100:8888"),
    username=os.getenv("WG_ADMIN_USER", "admin@wgportal.local"),
    api_token=os.getenv("WG_API_TOKEN", ""),
)


def main():
    user_id = "alice"

    try:
        CLIENT.users.get_by_id(user_id)
    except WgPortalError:
        print(f"User {user_id} not found")
        sys.exit(0)

    peers = CLIENT.peers.get_by_user(user_id)
    for peer in peers:
        print(f"Deleting peer: {peer['Identifier']} ({peer.get('DisplayName', '-')})")
        CLIENT.peers.delete(peer["Identifier"])

    CLIENT.users.delete(user_id)
    print(f"User {user_id} deleted")


if __name__ == "__main__":
    main()

File: lib/typescript/examples/02-offboard-user.ts
/**
 * Offboard a user: delete all their peers then delete the user account.
 *
 * Real case: employee leaves, HR triggers deprovisioning. All their
 * WireGuard access is revoked atomically.
 */

import { WgPortalClient, WgPortalError } from "../src";

const client = new WgPortalClient({
  baseUrl: "http://192.168.1.100:8888",
  username: "admin@wgportal.local",
  apiToken: process.env.WG_API_TOKEN!,
  logger: console,
});

async function offboardUser(userId: string) {
  // 1. Fetch all peers belonging to the user
  const peers = await client.peers.getByUser(userId);
  console.log(`Found ${peers.length} peer(s) for user ${userId}`);

  // 2. Delete each peer
  for (const peer of peers) {
    await client.peers.delete(peer.Identifier);
    console.log(`Deleted peer ${peer.Identifier} (${peer.DisplayName ?? "unnamed"})`);
  }

  // 3. Delete the user account
  await client.users.delete(userId);
  console.log(`User ${userId} deleted`);
}

offboardUser(process.argv[2] ?? "uid-1234567").catch((err) => {
  if (err instanceof WgPortalError) {
    console.error(`API error ${err.code}: ${err.message} — ${err.details}`);
    process.exit(1);
  }
  throw err;
});
